src/Engine/ProtoCore/SyntaxAnalysis/AssociativeAstVisitor.cs

```diff
@@ -0,0 +1,615 @@
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitFunctionCallNode(FunctionCallNode node)
+        {
+            return DefaultVisit(node);    
+        }
+
+        public virtual TResult VisitFunctionDotCallNode(FunctionDotCallNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitVarDeclNode(VarDeclNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitArgumentSignatureNode(ArgumentSignatureNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitCodeBlockNode(CodeBlockNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitClassDeclNode(ClassDeclNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitConstructorDefinitionNode(ConstructorDefinitionNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitFunctionDefinitionNode(FunctionDefinitionNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitIfStatementNode(IfStatementNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitInlineConditionalNode(InlineConditionalNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitBinaryExpressionNode(BinaryExpressionNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitUnaryExpressionNode(UnaryExpressionNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitRangeExprNode(RangeExprNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitExprListNode(ExprListNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitArrayNode(ArrayNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitImportNode(ImportNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitDynamicNode(DynamicNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitDynamicBlockNode(DynamicBlockNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitThisPointerNode(ThisPointerNode node)
+        {
+            return DefaultVisit(node);
+        }
+
+        public virtual TResult VisitDefaultArgNode(DefaultArgNode node)
+        {
+            return DefaultVisit(node);
+        }
+    }
+
+    public class AssociativeAstReplacer : AssociativeAstVisitor<AssociativeNode>
```

Comment:
Probably not needed, but will there be a scenario when we just want to replace only Associative nodes? Likewise, ImperativeAstReplacer is also left intact.